move/axelar_gateway/sources/gateway.move

```diff
@@ -186,13 +189,7 @@ entry fun approve_messages(
             proof,
         );
 
-    let mut i = 0;
-
-    while (i < messages.length()) {
-        self.approve_message(&messages[i]);
-
-        i = i + 1;
-    };
+    messages.length().do!(|i| self.approve_message(&messages[i]));
```

Comment:
```suggestion
    messages.map!(|message| self.approve_message(message));
```
see other available vector macros to use based on the situation. Change `approved_message` to take in the Message itself instead of the reference

Reply:
In this case, i think we want to iterate over the messages vector and calls `approve_message` for each message, instead of mapping them and return a new collection of data?

Reply:
Looks like they have `foreach` macro, use that instead

Reply:
i don't see `foreach` macro in [their doc](https://blog.sui.io/move-2024-macros-beta/). have they mentioned that anywhere?